### ISSUE TITLE
\\ in regex breaks highlights

### ISSUE THREAD

**Issue by zoffixznet** [perl6/atom-language-perl6#38]
I'm on 1.13.3.

Seems to be caused by `\\` in regex (seems to not be broken on GitHub... unsure why)

### Example Code
```perl6
sub MAIN ( Str $app, Str :$w = 'lib,bin') {
    my @watchlist = $w.split: /<!after \\> \,/;
    s/\\\,/,/ for @watchlist;

    say "Attempting to boot up the app";
    my $p = bootup-app $app;
    react {
        whenever watch-recursive(@watchlist.grep: *.IO.e) -> $e {
            say "Change detected [$e.path(), $e.event()]. Restarting app";
            $p.kill;
            $p = bootup-app $app;
        }
    }
}
```
### Leave this in. For internal use.
- [ ] Fixed in Master
- [ ] Fixed in Release
- [ ] Has Tests
- [ ] Passes Tests
![z](https://cloud.githubusercontent.com/assets/5747918/21682585/6cdd1ef4-d323-11e6-81a0-0b0cc95c4eb7.png)

**Comment by samcv** [comment]
It's not broken on Github because the old version of the highlighter doesn't highlight s/// at all!

I don't think the regex highlighting actually knows anything about what backslash even means, so I will have to set it to be something like what is used for quoted strings.

**Comment by samcv** [comment]
Oh my bad it's in the regex that the backslashes cause the problem, not s///. I will have to make <!after > know that one backslash is an escaped closing delimiter, but two backslashes is not

**Comment by samcv** [comment]
Actually it looks like it's another one of those cases where it thinks that this is an array of quoted words using the `< >` delimiters. Should be able to blacklist that before a /

**Comment by samcv** [comment]
This is fixed in v1.13.5 which has just been released. (The array word quoting issue. It does not register as regex though, but it doesn't break highlights). Closing.

If you think `$w.split: /<!after \\> \,/` can be unambiguously detected as regex, then reopen this and I will do that as well.
